multigeiger/speaker.cpp

```diff
@@ -111,43 +115,60 @@ void IRAM_ATTR isr_audio() {
   }
 }
 
-void IRAM_ATTR tick(bool tick) {
-  // make speaker tick and LED blink (or tock, no LED), called from ISR!
+void IRAM_ATTR tick(bool high) {
+  // high true: "tick" -> high frequency tick and LED blink
+  // high false: "tock" -> lower frequency tock, no LED
+  // called from ISR!
   portENTER_CRITICAL_ISR(&mux_audio);
-  isr_tick_sequence = tick ? tick_sequence : tock_sequence;
+
+  int *sequence;
+
+  // we expect speaker_tick or led_tick to change at any time,
+  // thus check it here and generate different sequences:
+  if (speaker_tick || led_tick) {
+    sequence = tick_sequence;
+    // "on"
+    sequence[0] = speaker_tick ? (high ? 5000000 : 1000000) : -1;  // frequency_mHz
+    sequence[1] = 1;  // volume
+    sequence[2] = led_tick ? (high ? 1 : -1) : -1;  // LED
+    sequence[3] = 4;  // duration_ms
+    // "off"
+    sequence[4] = speaker_tick ? 0 : -1;
+    sequence[5] = 0;
+    sequence[6] = led_tick ? (high ? 0 : -1) : -1;
+    sequence[7] = 0;  // END
+  } else
+    sequence = NULL;
```

Comment:
This is the important part. If the sequence is NULL, the ISR will be finished very quickly and not access any library code that resides in flash memory.